### PR TITLE
[docs] Clean ValidateRowModelControlGrid demo

### PR DESCRIPTION
**Pull request by alexfauquette**
- Add type to `preProcessEditCellProps`
- define `columns`/`rows` before using them

### CODE REVIEW

File: docs/data/data-grid/editing/ValidateRowModelControlGrid.tsx [hunk 1]
```diff
@@ -15,6 +16,39 @@ function validateEmail(email) {
...
+    },
+    {
+      field: 'dateCreated',
+      headerName: 'Date Created',
+      type: 'date',
+      width: 180,
+      editable: true,
+    },
+    {
+      field: 'lastLogin',
```
Comment (m4theushw) [inline docs/data/data-grid/editing/ValidateRowModelControlGrid.tsx:44]: The employee dataset doesn't this column. You can use `lastUpdated` to fake it.

```suggestion
      field: 'lastUpdated',
```

**Comment by alexfauquette** [comment]
Do you have an idea of why numbers appear in argos whereas they are not visible in the documentation example?

File: docs/data/data-grid/editing/ValidateRowModelControlGrid.tsx [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
-  randomTraderName,
-  randomUpdatedDate,
-} from '@mui/x-data-grid-generator';
+  DataGrid,
+  GridColumns,
+  GridPreProcessEditCellProps,
+} from '@mui/x-data-grid';
+import { useDemoData } from '@mui/x-data-grid-generator';
+
+const VISIBLE_FIELDS = ['name', 'email', 'dateCreated', 'lastLogin'];
```
Comment (m4theushw) [inline docs/data/data-grid/editing/ValidateRowModelControlGrid.tsx:10]: The last column is not visible.

```suggestion
const VISIBLE_FIELDS = ['name', 'email', 'dateCreated', 'lastUpdated'];
```

**Review by m4theushw — APPROVED** [review]
> Do you have an idea of why numbers appear in argos whereas they are not visible in the documentation example?

The name column adds a suffix if the name is already being used. You don't see it in the docs example because there we use random names.

https://github.com/mui/mui-x/blob/9e2c4b43436b970d269db2b623dbe2763182f601/packages/grid/x-data-grid-generator/src/columns/employees.columns.tsx#L50-L51

File: docs/data/data-grid/editing/ValidateRowModelControlGrid.tsx (hunk 2 above)
Comment (alexfauquette) [reply docs/data/data-grid/editing/ValidateRowModelControlGrid.tsx]: I realized I was looking at the wrong example in the doc 🙈

**Pull request merged**